src/main/java/com/google/sps/data/PersistentUserRepository.java

```diff
@@ -157,9 +180,7 @@ public User fetchUserWithId(String userId) {
     PreparedQuery results = getQueryFilterForId(userId);
     Collection<User> userProfiles = fetchUserEntities(results);
     if(userProfiles.size() == 1){
-        for (User userProfile : userProfiles) {
-            return userProfile;
-        }
+        return userProfiles.iterator().next();
     }
     User fake = new User(userId);
```

Comment:
What's the reasoning for returning a fake user with an id as a name?
If this is meant just to ensure you have a value to return, then it might make more sense to do something like:
`if(userProfiles.size() != 1) throw new Exception("User \"" + name + "\" does not exist");`
`return userProfiles.iterator().next()`

Reply:
i got an error about the `return userProfiles.iterator().next()` It says its unreachable since it should check for 1 size only

Reply:
Done